Code/GraphMol/Wrap/rdchem.cpp

```diff
@@ -78,28 +79,100 @@ struct PySysErrWrite : std::ostream, std::streambuf {
   void write(char c) {
     buffer += c;
     if (c == '\n') {
-      PySys_WriteStderr("%s", (prefix + buffer).c_str());
+      PySys_WriteStderr("%s%s", prefix.c_str(), buffer.c_str());
       buffer.clear();
     }
   }
 #endif
 };
 
-void RDLogError(const std::string &msg) {
+// std::ostream wrapper around Python's logging module
+struct PyLogStream : std::ostream, std::streambuf {
+  PyObject *logfn = nullptr;
+#ifdef RDK_THREADSAFE_SSS
+  static thread_local std::string buffer;
+#else
+  std::string buffer = "";
+#endif
+
+  PyLogStream(std::string level): std::ostream(this) {
+    PyObject *module = PyImport_ImportModule("logging");
+    PyObject *logger = nullptr;
+
+    if (module != nullptr) {
+      logger = PyObject_CallMethod(module, "getLogger", "s", "rdkit");
+      Py_DECREF(module);
+    }
+
+    if (logger != nullptr) {
+      logfn = PyObject_GetAttrString(logger, level.c_str());
+      Py_DECREF(logger);
+    }
+
+    if (PyErr_Occurred()) {
+      PyErr_Print();
+    }
+  }
+
+  ~PyLogStream() {
+    if (!_Py_IsFinalizing()) {
+      Py_XDECREF(logfn);
+    }
+  }
+
+  int overflow(int c) override {
+    write(c);
+    return 0;
+  }
+
+  void write(char c) {
+    if (logfn == nullptr) {
+      return;
+    }
+
+    if (c == '\n') {
+#ifdef RDK_THREADSAFE_SSS
+      PyGILStateHolder h;
+#endif
+      PyObject *result = PyObject_CallFunction(logfn, "s", buffer.c_str());
+      Py_XDECREF(result);
+      buffer.clear();
+    }
+    else {
+      buffer += c;
+    }
+  }
+};
+
+#ifdef RDK_THREADSAFE_SSS
+thread_local std::string PyLogStream::buffer;
+#endif
+
+void RDLogDebug(const std::string &msg) {
   NOGIL gil;
-  BOOST_LOG(rdErrorLog) << msg.c_str() << std::endl;
+  BOOST_LOG(rdDebugLog) << msg.c_str() << std::endl;
+}
+
+void RDLogInfo(const std::string &msg) {
+  NOGIL gil;
+  BOOST_LOG(rdInfoLog) << msg.c_str() << std::endl;
 }
 
 void RDLogWarning(const std::string &msg) {
   NOGIL gil;
   BOOST_LOG(rdWarningLog) << msg.c_str() << std::endl;
 }
 
+void RDLogError(const std::string &msg) {
+  NOGIL gil;
+  BOOST_LOG(rdErrorLog) << msg.c_str() << std::endl;
+}
+
 void WrapLogs() {
-  static PySysErrWrite debug("RDKit DEBUG: ");
-  static PySysErrWrite error("RDKit ERROR: ");
-  static PySysErrWrite info("RDKit INFO: ");
-  static PySysErrWrite warning("RDKit WARNING: ");
+  static PyErrStream debug("RDKit DEBUG: ");
```

Comment:
The intent of WrapLogs was to be a lot closer to what LogToPythonLogger does.  Also windows systems have a flushing issue with WrapLogs that may be fixed with LogToPythonLogger.

I expect WrapLogs should just call LogToPythonLogger.

Reply:
Okay!  My original plan was to leave all the existing functions untouched, but I agree that `LogToPythonLogger` is more useful / less surprising than the old `WrapLogs` behavior.  I have a commit coming soon that'll address this.

Do you have a link for the Windows flushing problems?  I doubt my changes will fix anything, 'cause they use the same underlying stream implementation, but I can take a look while I'm working on it.

Reply:
I think it'd be fine to postpone the windows problems to a separate PR